sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -61,15 +66,31 @@ export class ApplicationController extends BaseController {
         `Application id ${applicationId} was not found.`,
       );
     }
-
     return {
       data: application.data,
       id: application.id,
       applicationStatus: application.applicationStatus,
       applicationStatusUpdatedOn: application.applicationStatusUpdatedOn,
+      applicationNumber: application.applicationNumber,
+      applicationOfferingType: application.offering?.offeringIntensity,
+      applicationStartDate: dateString(application.offering?.studyStartDate),
+      applicationEndDate: dateString(application.offering?.studyEndDate),
+      applicationInstitutionName: application.location?.name,
+      applicationPIRStatus: application.pirStatus,
+      applicationAssessmentStatus: application.assessmentStatus,
+      applicationCOEStatus: application.coeStatus,
+      applicationFormName: application.programYear.formName,
+      applicationProgramYearID: application.programYear.id,
+      applicationPIRDeniedReason:
+        application.pirDeniedReasonId?.id === PIR_DENIED_REASON_OTHER_ID
+          ? application.pirDeniedOtherDesc
+          : application.pirDeniedReasonId?.reason,
+      applicationCOEDeniedReason:
+        application.coeDeniedReason?.id === COE_DENIED_REASON_OTHER_ID
```

Comment:
Generally writing transformation in separate private method(at least) or  transformation Util method is more encouraged IMO. 

1. It will reduce the lines of code in controller method.
2. If there is need for re-usability, it will be easier.